src/main/java/fruitshop/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        Storage storage = new Storage();
```

Comment:
Creating a new instance of Storage inside the handler method is incorrect. The operation should work with the existing Storage instance that holds the current state of the fruits in the system.